frontend/gateway/client/client.go

```diff
@@ -16,6 +17,61 @@ type Client interface {
 	ResolveImageConfig(ctx context.Context, ref string, opt llb.ResolveImageConfigOpt) (digest.Digest, []byte, error)
 	BuildOpts() BuildOpts
 	Inputs(ctx context.Context) (map[string]llb.State, error)
+	NewContainer(ctx context.Context, req NewContainerRequest) (Container, error)
+}
+
+// NewContainerRequest encapsulates the requirements for a client to define a
+// new container, without defining the initial process.
+type NewContainerRequest struct {
+	Mounts       []Mount
+	NetMode      pb.NetMode
+	SecurityMode pb.SecurityMode
```

Comment:
I think `SecurityMode` should be per process, not per container. https://github.com/opencontainers/runtime-spec/blob/master/specs-go/config.go#L50-L60

Reply:
Yeah, sounds right.  I have updated: https://github.com/moby/buildkit/pull/1627/commits/ea3bc24b7c91abe5daf120b35bddd001a6e9865a